scipy/linalg/_special_matrices.py

```diff
@@ -286,18 +308,21 @@ def leslie(f, s):
     """
     f = np.atleast_1d(f)
     s = np.atleast_1d(s)
-    if f.ndim != 1:
-        raise ValueError("Incorrect shape for f.  f must be 1D")
-    if s.ndim != 1:
-        raise ValueError("Incorrect shape for s.  s must be 1D")
-    if f.size != s.size + 1:
-        raise ValueError("Incorrect lengths for f and s.  The length"
-                         " of s must be one less than the length of f.")
-    if s.size == 0:
+
+    if f.shape[-1] != s.shape[-1] + 1:
+        raise ValueError("Incorrect lengths for f and s. The length of s along "
+                         "the last axis must be one less than the length of f.")
+    if s.shape[-1] == 0:
         raise ValueError("The length of s must be at least 1.")
 
+    n = f.shape[-1]
+
+    if f.ndim > 1 or s.ndim > 1:
+        from scipy.stats._resampling import _vectorize_statistic
```

Comment:
We can eliminate this import by using the decorator from gh-21462.

Reply:
I assume we can now do this?

Reply:
Yeah I think so